tsinfer/inference.py

```diff
@@ -377,10 +377,16 @@ def infer(
     )
     if record_provenance:
         tables = inferred_ts.dump_tables()
-        record = provenance.get_provenance_dict(
-            command="infer",
-            mismatch_ratio=mismatch_ratio,
-        )
+        params = {  # TODO: maybe record recombination rate (which could be a RateMap)
+            "mismatch_ratio": mismatch_ratio,
+            "path_compression": path_compression,
+            "precision": precision,
+        }
+        if simplify is None:
+            params["post_process"] = post_process
+        else:
+            params["simplify"] = simplify
+        record = provenance.get_provenance_dict(command="infer", **params)
```

Comment:
This seems like a backward step duplicating the logic for dealing with simplify.

Would it be better to pass everything to ``get_provenance_dict`` and sort out any messiness that's needed in there, once?

Reply:
Yeah, it is a bit messy, I thought it might be overkill to keep the backwards compatibility, but then realised that we would potentially be reporting *wrong* parameters, as `simplify` now sets the post_process variable to True.

I've now done what you suggested and stuck the logic in get_provenance_dict, thanks for the suggestion.